### PR TITLE
LiDAR Detection (Issue #368)

### PR DESCRIPTION
**Pull request by MtGuerenS**
## Description
Implemented a lidar detection node that combines the [MMDetection3D](https://mmdetection3d.readthedocs.io/en/latest/) and the [ComplexYOLOv4](https://github.com/maudzung/Complex-YOLOv4-Pytorch) models for the [Object classification (Lidar)](https://github.com/Nova-UTD/navigator/issues/368) Issue.

## Usage
To test the new feature do the following:
1. Click [here](https://drive.google.com/drive/folders/16zuyjh0c7iiWRSNKQY7CnzXotecYN5vc) and download the `complex_yolov4_mse_loss.pth` file
2. Move `complex_yolov4_mse_loss.pth` into the directory `navigator/data/perception/checkpoints/complex_yolov4_mse_loss.pth`
3. cd into `/navigator`
4. uncomment `# *object3d_launch_entities,` in `launches/launch.perception.py`
5. run `docker compose run navigator_carla`
6. run `colcon build --symlink-install --packages-select lidar_detection && source install/setup.bash`
7. run `ros2 launch launches/launch.vehicle.py`
8. open new ternminal with pwd in `/navigator`
9. run `docker compose run navigator_carla`
10. run `ros2 bag play /bags/2024/bag_2024-02-09_Demo/` or other bag

### CODE REVIEW

File: data/navigator_default.rviz
Comment (danielv012) [inline data/navigator_default.rviz:1]: What's with all the changes to the default rviz config?

File: docs/Perception/static_occupancy_node.md
Comment (danielv012) [inline docs/Perception/static_occupancy_node.md:1]: thx for covering this

File: src/perception/lidar_detection/lidar_detection/complex_yolov4/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+The PyTorch Implementation based on YOLOv4 of the paper: [Complex-YOLO: Real-time 3D Object Detection on Point Clouds](https://arxiv.org/pdf/1803.06199.pdf)
+
+Most files are taken (some modified) from the GitHub Repo: [https://github.com/maudzung/Complex-YOLOv4-Pytorch/tree/master](https://github.com/maudzung/Complex-YOLOv4-Pytorch/tree/master)
+
+## Installation and Setup
+
+Follow the instrucitons below to ensure the model is setup properly:
+1. Click [here](https://drive.google.com/drive/folders/16zuyjh0c7iiWRSNKQY7CnzXotecYN5vc) and download the `complex_yolov4_mse_loss.pth` file
+2. Move `complex_yolov4_mse_loss.pth` into the directory `navigator/data/perception/checkpoints/complex_yolov4_mse_loss.pth`
```
Comment (danielv012) [inline src/perception/lidar_detection/lidar_detection/complex_yolov4/README.md:11]: Is it possible to change the path to navigator/data/perception/models? It's where we've historically kept our model weight files

File: data/navigator_default.rviz
Comment (danielv012) [reply data/navigator_default.rviz:1]: Was this a rebase?

File: launches/launch.object3d.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        package='visualizer',
+        executable='object_visualizer_node',
+        name='object_visualizer_node',
+        parameters=[
+            {'topic': '/tracked/objects3d'}
+        ]
+    )
+
+    return LaunchDescription([
+        complex_yolo_model,
```
Comment (danielv012) [inline launches/launch.object3d.py:68]: can comment either of the two out right now since we don't want both running at the same time

**Review by danielv012 — CHANGES_REQUESTED** [review]
Everything looks awesome. Just a few comments I left

File: data/navigator_default.rviz
Comment (MtGuerenS) [reply data/navigator_default.rviz:1]: I added a group for object detection/tracking, and the rest of the changes are just the position I had the camera at. Here is the most important part of the changes:

```
- Class: rviz_common/Group
      Displays:
        - Class: rviz_default_plugins/MarkerArray
          Enabled: true
          Name: Detected
          Namespaces:
            object_bounding_box: true
            object_tag: true
          Topic:
            Depth: 5
            Durability Policy: Volatile
            History Policy: Keep Last
            Reliability Policy: Reliable
            Value: /viz/detected/objects3d
          Value: true
        - Class: rviz_default_plugins/MarkerArray
          Enabled: false
          Name: Tracked
          Namespaces:
            {}
          Topic:
            Depth: 5
            Durability Policy: Volatile
            History Policy: Keep Last
            Reliability Policy: Reliable
            Value: /viz/tracked/objects3d
          Value: false
      Enabled: true
      Name: 3D Objects
```

File: src/perception/lidar_detection/lidar_detection/complex_yolov4/README.md (hunk 1 above)
Comment (MtGuerenS) [reply src/perception/lidar_detection/lidar_detection/complex_yolov4/README.md:11]: I'll move both Ragib's and my files to that directory then

**Comment by ragibarnab** [comment]
Hi Gueren. Please don't hardcode the paths to model configs/weights, change it so they pass them through ros params. There's a reason I used MMDetection. That is, you can take any state-of-the-art model (a lot of them better than ComplexYOLO) from MMDetection package and use it for inference. By hardcoding the configs/weights the key functionality of my package is lost. 

Thanks,
Rae

P.S. in the future just communicate with others if you are going to edit and remove their code. Communication is key, y'know? I'm a nice guy, I promise, just reach me out on teams. Like really, our packages should've been split instead of combined the way it is to prevent any issues.

**Review by danielv012 — APPROVED** [review]

**Pull request merged**